--- a/storage/myisam/rt_index.cc
+++ b/storage/myisam/rt_index.cc
@@ -685,7 +685,6 @@
                             stPageList *ReinsertList, int level) {
   uchar *k;
   uchar *last;
-  ulong i;
   uint nod_flag;
   uchar *page_buf;
   int res;
@@ -704,7 +703,7 @@
   k = rt_PAGE_FIRST_KEY(page_buf, nod_flag);
   last = rt_PAGE_END(page_buf);
 
-  for (i = 0; k < last; k = rt_PAGE_NEXT_KEY(k, key_length, nod_flag), ++i) {
+  for (; k < last; k = rt_PAGE_NEXT_KEY(k, key_length, nod_flag)) {
     if (nod_flag) {
       /* not leaf */
       if (!rtree_key_cmp(keyinfo->seg, key, k, key_length, MBR_WITHIN)) {
--- a/storage/myisam/sp_key.cc
+++ b/storage/myisam/sp_key.cc
@@ -54,7 +54,6 @@
   uint dlen;
   uchar *dptr;
   double mbr[SPDIMS * 2];
-  uint i;
 
   keyseg = &keyinfo->seg[-1];
   pos = record + keyseg->start;
@@ -67,7 +66,7 @@
   }
   sp_mbr_from_wkb(dptr + 4, dlen - 4, SPDIMS, mbr); /* SRID */
 
-  for (i = 0, keyseg = keyinfo->seg; keyseg->type; keyseg++, i++) {
+  for (keyseg = keyinfo->seg; keyseg->type; keyseg++) {
     uint length = keyseg->length, start = keyseg->start;
     double val;
 
